src/main/java/org/example/homework4/LeapYear.java

```diff
@@ -0,0 +1,14 @@
+package org.example.homework4;
+
+public class LeapYear {
+    public boolean isLeapYear(int year) {
+        if (year < 1 || year > 9999) {
+            return false;
+        }
+        if (year % 4 == 0 && year % 100 != 0 || year % 100 == 0 && year % 400 == 0) {
```

Comment:
this is correct but i suggest use more `()`. e.g. 
`((year % 4 == 0 && year % 100 != 0) || (year % 400 == 0))` - this expression is a bit more readable